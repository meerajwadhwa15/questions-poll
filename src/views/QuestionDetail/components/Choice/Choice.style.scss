.choice {
    display: flex;
    justify-content: space-between;
    padding: 10px;
    border: 1px solid #000;

    > div {
        width: 50%;
        text-align: left;
    }
}
