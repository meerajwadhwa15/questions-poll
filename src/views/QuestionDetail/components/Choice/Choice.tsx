import React, { FunctionComponent } from "react";
import { Choice } from "@/types/questions";

export type Props = {
  choice: Choice;
};

const List: FunctionComponent<Props> = ({
  choice: { choice, votes, url }
}) => (
    <div className="choice">
      <div>Choice: <strong>{choice}</strong></div>
      <div>Votes: <strong>{votes}</strong></div>
    </div>
  );

export default List;
