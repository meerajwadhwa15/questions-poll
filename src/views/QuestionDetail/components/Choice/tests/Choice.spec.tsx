import React from "react";
import Choice from "../Choice";
import { create } from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";


const choice = {
    choice: "Swift",
    votes: 39,
    url: "/questions/1/choices/1"
};

const props = {
    choice,
};

test("Matches the snapshot with Questions Choice", () => {
    const component = create(
        <Choice {...props} />
    );
    expect(component.toJSON()).toMatchSnapshot();
});
