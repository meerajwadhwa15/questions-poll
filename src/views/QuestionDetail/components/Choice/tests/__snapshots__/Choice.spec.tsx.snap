// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches the snapshot with Questions Choice 1`] = `
<div
  className="choice"
>
  <div>
    Choice: 
    <strong>
      Swift
    </strong>
  </div>
  <div>
    Votes: 
    <strong>
      39
    </strong>
  </div>
</div>
`;
